File: uri 1179 - Array Fill IV.c
#include <stdio.h>
#include <stdlib.h>

void parp(int j,int pa[]){
    int i;
    for(i=0;i<j;i++)
        printf("par[%d] = %d\n",i,pa[i]);
}
void imparp(int z ,int im[]){
    int i;
    for(i=0;i<z;i++)
        printf("impar[%d] = %d\n",i,im[i]);
}

int main()
{
    int i,j=0,z=0,k,l=15,pa[5],im[5];
    while(l>0){
        l--;
        scanf("%d",&k);
        if(k%2 == 0){
            pa[j]=k;
            j++;
            if(j==5){
               parp(j,pa);
               j=0;
            }
        }
        else{
            im[z]=k;
            z++;
            if(z==5){
                imparp(z,im);
                z=0;
            }
        }
    }
    imparp(z,im);
    parp(j,pa);
    return 0;
}

File: uri 1179 - Array Fill IV.py
par = []
impar = []
for i in range(15):
    v = int(input())
    if v % 2 ==0:
        par.append(v)
    else:
        impar.append(v)
    if len(impar) == 5:
        for j in range(5):
            print("impar[%d] = %d"%(j,impar[j]))
        impar = []
    if len(par) == 5:
        for j in range(5):
            print("par[%d] = %d"%(j,par[j]))
        par = []
for j in range(len(impar)):
    print("impar[%d] = %d"%(j,impar[j]))
for j in range(len(par)):
    print("par[%d] = %d"%(j,par[j]))
